src/v2/view-builder/internals/BaseForm.js

```diff
@@ -111,39 +111,45 @@ export default Form.extend({
       }
     }
   },
-
+  
   /*
-   * Whether an form error message should be rendered as the default BaseForm error banner or not
-   * if this function returns true, BaseForm won't render the error banner after form submit. Instead,
-   * the form needs to override showCustomErrorCallout with the logic to render that error message
-   */
-  isErrorCalloutCustomized() {
-    return false;
-  },
-
-  /*
-   * Renders an error message (if any) after form is submitted
-   * This should be overriden if isErrorCalloutCustomized() returns true
-   */
-  showCustomErrorCallout() {
-    this.$('.o-form-error-container').addClass('o-form-has-errors');
-  },
+  * Views should override this function to render custom error callouts for invalid form actions.
+  */
+  showCustomFormErrorCallout: _.identity,
 
   /*
-   * Renders the contents of messages object (if any) during initialize
+   * Renders the contents of messages object (if any, on error) during initialize
    * This function is called during Form.initialize, and will display
-   * messages when the form renders.
-   * Note: any errors happening after the form is rendered won't use this function,
-   * this is only used during initialize
+   * messages when the form reloads.
+   * Note: Any user action related form errors handled by FormController.showFormErrors
    */
-  showMessages() {
-    // render messages as text
-    const messagesObjs = this.options.appState.get('messages');
-    if (messagesObjs?.value.length) {
-      const content = messagesObjs.value.map((messagesObj) => {
-        return messagesObj.message;
+  showMessages(options) {
+    const messages = this.options.appState.get('messages') || {};
+    const errContainer = '.o-form-error-container';
+    if (Array.isArray(messages.value)) {
+      this.add('<div class="ion-messages-container"></div>', errContainer);
+      messages.value.forEach(obj => {
+        if(!obj?.class || obj.class === 'INFO') {
```

Comment:
why are we including `info` as well here? We have a callout type `info` that we can use if we now have this `class` attribute in the message. Your changes below already will take care of that

Reply:
anything that is not `ERROR` or `WARN` show up as plain text messages.. We don't show `INFO` callouts in widget.. but we do have messages tagged as `INFO`..